packages/mui-base/src/FormControlUnstyled/FormControlUnstyled.types.ts

```diff
@@ -98,3 +98,5 @@ export type FormControlUnstyledRootSlotProps = {
   className?: string;
   ownerState: FormControlUnstyledOwnerState;
 };
+
+export type UseFormControlUnstyledContextReturnValue = FormControlUnstyledState | undefined;
```

Comment:
You need to declare `UseFormControlUnstyledContextReturnValue` as an interface:
```suggestion
export interface UseFormControlUnstyledContextReturnValue extends FormControlUnstyledState {};
```

And then in `useFormControlUnstyledContext` hook return a union type with `undefined` type member:
```
export default function useFormControlUnstyledContext(): UseFormControlUnstyledContextReturnValue | undefined
```

----

Then you add the JSDoc descriptions for all the props and the docs will get generated.